storage/pipeline/fsm_events.go

```diff
@@ -182,6 +182,8 @@ func (evt SectorSealPreCommit1Failed) FormatError(xerrors.Printer) (next error)
 func (evt SectorSealPreCommit1Failed) apply(si *SectorInfo) {
 	si.InvalidProofs = 0 // reset counter
 	si.PreCommit2Fails = 0
+
+	si.PreCommit1Fails++
```

Comment:
Where do you set/reset this counter?

Reply:
It starts at zero because it's Go, it's also never reset, because we never do anything that could fix the (possibly corrupted) inputs to PC1

Basically it only ever counts up, until 3 retries, when the sector is removed

Reply:
And its not possible to go from a successful PC1 after 2 tries, go to subsequent stages where some failure occurs and end up back at PC1? Since your counter won't be reset

Reply:
Correct, the idea is that if PC1 has already failed a few times, the problem is probably with its inputs, and if we're not doing anything to fix the inputs, there's probably no hope for the sector